cmd/manager/main.go

```diff
@@ -85,8 +85,8 @@ func main() {
 
 	namespace, err := k8sutil.GetWatchNamespace()
 	if err != nil {
-		log.Error(err, "Failed to get watch namespace")
-		os.Exit(1)
+		// Default to watching all namespaces
+		namespace = ""
 	}
```

Comment:
Alternative implementation could be:

```Golang
	namespace := ""
	if os.Getenv(k8sutil.WatchNamespaceEnvVar) != "" {
		var err error
		namespace, err = k8sutil.GetWatchNamespace()
		if err != nil {
			log.Error(err, "Failed to get watch namespace")
			os.Exit(1)
		}
	}
```

so we still error in case an invalid value for `WATCH_NAMESPACE` is set.

Reply:
I like this one better, have pushed the change. Let me know if you think otherwise.

Reply:
Actually, `k8sutil.GetWatchNamespace()` only fails when the envvar is not set. It doesn't do any additional validation. So the error case is dead code and the full snippet is equivalent to the below code:
```Golang
namespace := os.Getenv(k8sutil.WatchNamespaceEnvVar)
```

I've pushed that change.